src/Grids/input_validation.jl

```diff
@@ -178,6 +178,9 @@ validate_unit_vector(ê::NegativeZDirection) = ê
 function validate_unit_vector(ê)
     length(ê) == 3 || throw(ArgumentError("unit vector must have length 3"))
 
+    # ensures that all components of ê are of the same type
+    ê = Tuple([ê...])
```

Comment:
this is to avoid having `ê :: Tuple{Int64, Float64, Float64}`. Ideally I think we'd like to ensure all elements are of the type of `eltype(grid)` but at the point the buoyancy model is constructed the grid is not known.

cc @glwagner